pkg/util/validation/validation.go

```diff
@@ -59,6 +60,9 @@ func ValidateClusterAffinities(affinities []policyv1alpha1.ClusterAffinityTerm,
 
 	affinityNames := make(map[string]bool)
 	for index, term := range affinities {
+		for _, err := range validation.IsDNS1123Subdomain(term.AffinityName) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(index), term.AffinityName, err))
```

Comment:
I'm afraid [RFC1123 Label Names](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names) is a little bit too strict, which only accepts lowercase alphanumeric characters.
@XiShanYongYe-Chang Please take a look.

To be honest, I really don't want to add such a restriction that we should do in the last release, for people already using this feature, it'd be a challenge to upgrade.

Reply:
Compared to [RFC1123 Label Names](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names), how about the following?

https://github.com/kubernetes/kubernetes/blob/b1f901acf44f8adf69759db434690c2a8bdd616b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L31-L34

```go
const qnameCharFmt string = "[A-Za-z0-9]"
const qnameExtCharFmt string = "[-A-Za-z0-9_.]"
const qualifiedNameFmt string = "(" + qnameCharFmt + qnameExtCharFmt + "*)?" + qnameCharFmt
const qualifiedNameErrMsg string = "must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character"
```

Reply:
ok

Reply:
/cc @RainbowMango @XiShanYongYe-Chang Please take a look.